Replaced deprecated API endpoints and added channel list paging

shamarashid opened the pull request:
Replaced deprecated channels endpoints with conversations endpoints. Slack announcement  https://api.slack.com/changelog/2020-01-deprecating-antecedents-to-the-conversations-api#:~:text=The%20final%20date%20for%20existing,All%20channels. 

Added ability to iterate over large list of channels using paging and cursors.

Commit:
Slack channels endpoints deprecated in Feb, 2021; repalced by conversations endpoints

Commit:
Handled large list of channels by adding pagination capabilities using cursor

nilsreichardt commented on slack_autoarchive.py:
```diff
@@ -78,16 +78,28 @@ def slack_api_http(
+                        'Need to add bot to channel described by ' + str(payload)
+                    )
+                    self.join_channel(payload['channel'])
+                    return response.json()
```
```suggestion
                    return self.slack_api_http(api_endpoint, payload, method)
```

TL;DR: Without this fix, the bot will archive **EVERY** channel

If you just return the `response` the following will happen:
1. Bot tries to read history -> rejected
2. Bot joins channel
3. Return `repsonse`, but in `response` is the message "not_in_channel"
4. Because no messages are in the request -> bot will archive the channel

shamarashid replied:
@nilsreichardt : Thanks for catching the bug. Suggested changes applied.

shamarashid committed:
Bug fix - resubmit channel history request after joining a channel.

Without this fix, the channel will be archived right after the bot joins the channel since channel history has no messages.

Co-authored-by: Nils Reichardt <me@nils.re>
